File: Java/Memory.java
// derived class memory
class Memory extends Hardware{
	
	/* atribut private */
	private int frequency;
	private int memorySize;
	private String supportsCuda;

	/* konstruktor */
	public Memory(){}

	public Memory(int frequency, int memorySize, String supportsCuda){
		this.frequency = frequency;
		this.memorySize = memorySize;
		this.supportsCuda = supportsCuda;
	}

	/* setter dan getter untuk atribut frequency */
	public int getFrequency(){
		return frequency;
	}

	public void setFrequency(int frequency){
		this.frequency = frequency;
	}

	/* setter dan getter untuk atribut memorySize */
	public int getMemorySize(){
		return memorySize;
	}

	public void setMemorySize(int memorySize){
		this.memorySize = memorySize;
	}

	/* setter dan getter untuk atribut supportsCuda */
	public String getSupportsCuda(){
		return supportsCuda;
	}

	public void setSupportsCuda(String supportsCuda){
		this.supportsCuda = supportsCuda;
	}

	/* method untuk mencetak output atribut dari class Memory */
	public void printMemory(){
		System.out.println("Frekuensi      : " + this.getFrequency() + " MHz");
		System.out.println("Memori         : " + this.getMemorySize() + " GB");
		System.out.println("Mendukung Cuda : " + this.getSupportsCuda());
	}
}

File: Python/Memory.py
# import class
from Product import Product
from Hardware import Hardware

class Memory(Hardware):
	
	# atribut private
	__frequency = 0
	__memorySize = 0
	__supportsCuda = ""

	# konstruktor
	def __init__(self):
		self.__frequency = 0
		self.__memorySize = 0
		self.__supportsCuda = ""
	
	# setter dan getter untuk atribut frequency
	def setFrequency(self, frequency):
		self.__frequency = frequency

	def getFrequency(self):
		return self.__frequency

	# setter dan getter untuk atribut memorySize
	def setMemorySize(self, memorySize):
		self.__memorySize = memorySize
		
	def getMemorySize(self):
		return self.__memorySize

	# setter dan getter untuk atribut supportsCuda
	def setSupportsCuda(self, supportsCuda):
		self.__supportsCuda = supportsCuda

	def getSupportsCuda(self):
		return self.__supportsCuda

	# method untuk mencetak output atribut dari class Memory
	def printMemory(self):
		print("Frekuensi      : {} MHz".format(self.getFrequency()))
		print("Memori         : {} GB".format(self.getMemorySize()))
		print("Mendukung Cuda : " + self.getSupportsCuda())